File: extensions/instagram/src/main/java/app/revanced/extension/instagram/misc/share/privacy/SanitizeSharingLinksPatch.java
package app.revanced.extension.instagram.misc.share.privacy;

import app.revanced.extension.shared.privacy.LinkSanitizer;

@SuppressWarnings("unused")
public final class SanitizeSharingLinksPatch {
    private static final LinkSanitizer sanitizer = new LinkSanitizer("igsh");

    /**
     * Injection point.
     */
    public static String sanitizeSharingLink(String url) {
        return sanitizer.sanitizeUrlString(url);
    }
}

File: patches/src/main/kotlin/app/revanced/patches/instagram/misc/privacy/SanitizeSharingLinksPatch.kt
package app.revanced.patches.instagram.misc.privacy

import app.revanced.patcher.patch.bytecodePatch

@Deprecated(
    "Patch was moved to a different package",
    ReplaceWith("app.revanced.patches.instagram.misc.share.privacy.sanitizeSharingLinksPatch")
)
@Suppress("unused")
val sanitizeSharingLinksPatch = bytecodePatch {
    dependsOn(app.revanced.patches.instagram.misc.share.privacy.sanitizeSharingLinksPatch)
}
